### PR TITLE
feat: settings for speaker

### PR DESCRIPTION
**Pull request by phollome**
## List issues that are affected

#314 

- [x] Tests

### CODE REVIEW

File: app/routes/event/$slug/settings/speakers/add-speaker.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+      await action({ request, context: {}, params: {} });
+    } catch (error) {
+      const response = error as Response;
+      expect(response.status).toBe(404);
+
+      const json = await response.json();
+      expect(json.message).toBe("Event not found");
+    }
+  });
+
```
Comment (nac62116) [inline app/routes/event/$slug/settings/speakers/add-speaker.spec.tsx:80]: Duplicate tests "authenticated user" and "not privileged user"

File: app/routes/event/$slug/settings/speakers/remove-speaker.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+      await action({ request, context: {}, params: {} });
+    } catch (error) {
+      const response = error as Response;
+      expect(response.status).toBe(404);
+
+      const json = await response.json();
+      expect(json.message).toBe("Event not found");
+    }
+  });
+
```
Comment (nac62116) [inline app/routes/event/$slug/settings/speakers/remove-speaker.spec.tsx:79]: Duplicate Tests "authenticated user" and "not privileged user"

**Review by nac62116 — CHANGES_REQUESTED** [review]

**Comment by nac62116** [comment]
Nice one! Only found the duplicate tests. The rest looks superb! 🚀 

One last question:

What feels better concerning the nested routes?
- Working with an <Outlet> and defining the add/remove forms inside the child routes (see organization/settings/team)
- ...or working with only one index function inside the parent route and only outsourcing the action function (like you did here)

**Comment by phollome** [comment]
@nac62116 I've removed the duplicate tests. 👍

It feels better to have the forms in one route and just outsource the actions of the forms. Having forms in the action routes confused me this time. 😊 I've linked the question to refactoring/discussion issue #337.

**Review by nac62116 — APPROVED** [review]
Nice one 🏎️

**Pull request merged**